cmd/protoc-gen-go-grpc/grpc.go

```diff
@@ -111,6 +112,8 @@ func generateFile(gen *protogen.Plugin, file *protogen.File) *protogen.Generated
 	}
 	filename := file.GeneratedFilenamePrefix + "_grpc.pb.go"
 	g := gen.NewGeneratedFile(filename, file.GoImportPath)
+	// genStandaloneComments(g, file.Desc, int32(genid.FileDescriptorProto_Syntax_field_number))
+	genStandaloneComments(g, file.Desc, 12)
```

Comment:
What is this constant?  And please do not submit commented out code

Reply:
I made this PR as a request from the issue I reported #5530. Please review that discussion as to what this PR is for and why I made it.

I left the commented-out code to show what the magic number 12 meant. 
`12` is `genid.FileDescriptorProto_Syntax_field_number` but that constant is in an internal package.

I made this PR hoping you all could find a better solution to including the header comments from the .proto file, since you know the code better.

Reply:
We don't actually have that much expertise with the code generator internals, so it would take some ramp-up time for us to fully understand what's going on behind the scenes and try to determine a better way of doing this.

Maybe @neild can help with this particular issue of this magic constant, apparently needed by a public API, but defined in an internal package?  It looks like it's needed here:

```go
loc := desc.SourceLocations().ByPath(protoreflect.SourcePath{n /* 12 A.K.A genid.FileDescriptorProto_Syntax_field_number */})
```

`loc` is then used to get `LeadingDetachedComments` and `LeadingComments` so they can be added to the generated code.  But if it is only defined in an internal package, then I don't think we should be using it.

Reply:
Protobuf descriptor messages identify source comments using a "path", which is an array of numbers identifying the location where the comment was found:
https://github.com/protocolbuffers/protobuf/blob/v21.5/src/google/protobuf/descriptor.proto#L767

The Go `protoreflect` package uses the same location paths as the descriptor messages.

"12" is the field number of the `FileDescriptorProto.syntax` field, so a path of `[12]` here identifies the comments attached to that field.

I'd write this here as:

```
const fileDescriptorProto_syntax_field_number = 12
genStandaloneComments(g, file.Desc, fileDescriptorProto_syntax_field_number)

g.P("// Code generated by protoc-gen-go-grpc. DO NOT EDIT.")
g.P("// versions:")
// ...

const fileDescriptorProto_package_field_number = 2
genStandaloneComments(g, file.Desc, fileDescriptorProto_package_field_number)
g.P()
g.P("package ", file.GoPackageName)
// ...
```

This includes both comments attached to the `syntax` and `package` statements, in the same format used by `protoc-gen-go`.

(You could also write this to look up the relevant field numbers from the `FileDescriptorProto` descriptor, but inlining the constants is simpler.)

Reply:
Added constants, using camel case instead of snake case.